examples/log-ingestion/log_ingestion_demo_guide.md

```diff
@@ -0,0 +1,136 @@
+        username: admin
+        password: admin
+        index: grok-prepper
+```
+   
+4. Create the following `data-prepper-config.yaml`
+
+```yaml
+ssl: false
+```
+
+5. From the root of the data prepper repo, 
+   run the Data Prepper jar with the `pipeline.yaml` and `data-prepper-config.yaml` as command line arguments.
+   
+```
+java -jar data-prepper-core/build/libs/data-prepper-core-1.2.0-SNAPSHOT.jar /full/path/to/pipeline.yaml /full/path/to/data-prepper-config.yaml
+```
+
+If you see an error that looks like this: `Caused by: java.lang.RuntimeException: Connection refused`, then that probably means you don't have OpenSearch running locally. 
+Go [here](https://opensearch.org/downloads.html) to do so before moving on to the next step of this guide.
+
+### FluentBit Setup
+
+The FluentBit setup includes a `docker-compose.yaml`, a `fluent-bit.conf`, and a log file which corresponds to where FluentBit will look for logs.
+
+1. Create the following `docker-compose.yaml`. This `docker-compose.yaml` will pull the FluentBit Docker image, and will mount your `fluent-bit.conf` and log file to this Docker image. In this case, the log file is named `test.log`.
+
+```yaml
+version: "3.7"
+
+services:
+  fluent-bit:
+    image: fluent/fluent-bit
+    volumes:
+      - ./fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf
+      - ./test.log:/var/log/test.log
+```
+
+2. Create the following `fluent-bit.conf`. This config will tell FluentBit to tail the `/var/log/test.log` file for logs, and uses the FluentBit http output plugin to forward these logs to the http source of Data Prepper, which runs by default on port 2021.
+
+```
+[INPUT]
+  name                  tail
+  refresh_interval      5
+  path                  /var/log/test.log
+  read_from_head        true
+
+[OUTPUT]
+  Name http
+  Match *
+  Host host.docker.internal
+  Port 2021
+  URI /log/ingest
+  Format json
+```
+
+3. Create an empty file named `test.log`. This file can be named whatever you like, but the `docker-compose.yaml` will need to be updated accordingly.
+
+
+4. Now that you have the `docker-compose.yaml`, `fluent-bit.conf`, and the `test.log` files, FluentBit is ready for log ingestion. Start FluentBit with
+
+```
+docker-compose up
+```
+
+### Apache Log Generator
+
+In order to simulate an application generating logs, a simple python script will be used. This script only runs with python 2. You can download this script by running
+
+```
+git clone https://github.com/graytaylor0/Fake-Apache-Log-Generator.git
+```
+
+Note the requirements in the README of the Apache Log Generator. You must have Python 2.7 and you must run 
+```
+pip install -r requirements.txt
+```
+
+to install the necessary dependencies.
+
+Run the apache log generator python script so that it sends an apache log to the `test.log` file from the fluent-bit `docker-compose.yaml` every 2 seconds. 
+
+```
+python apache-fake-log-gen.py -n 0 -s 2 -l "CLF" -o "LOG" -f "/full/path/to/test.log"
+```
+
```

Comment:
It might be nice to call out that the user can now see fluentbit and data prepper processing logs. I encountered a docker networking error the first time I went through this. I had to look at the fluentbit out put to figure out the issue. Including the follow outputs will probably be helpful:

Successful fluentbit output
```
fluent-bit_1  | [2021/10/30 17:16:39] [ info] [output:http:http.0] host.docker.internal:2021, HTTP status=200
```

Successful DP output:
```
2021-10-30T12:17:17,474 [grok-pipeline-prepper-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker -  grok-pipeline Worker: Processing 2 records from buffer
```